### ISSUE TITLE
Detection loss should not divide all element

### ISSUE THREAD

**Issue by northeastsquare** [argman/EAST#228]
In model.py, the last line:
return tf.reduce_mean(L_g * y_true_cls * training_mask) + classification_loss
Maybe it should write like this:
return tf.sum(L_g * y_true_cls * training_mask)/ tf.count_nonzero(y_true_cls * training_mask) + classification_loss

**Comment by caoyangcr7** [comment]
Yes, I agree with you. But I changed this loss function only by removing the y_true_cls, then It can converge in my dataset. As a result , the problem that the model loss is 0.01 which never changed is also solved.

**Comment by xinghalo** [comment]
> Yes, I agree with you. But I changed this loss function only by removing the y_true_cls, then It can converge in my dataset. As a result , the problem that the model loss is 0.01 which never changed is also solved.

I find the classification_loss is never change,  too.
